### ISSUE TITLE
Weekly Community Check-In #71 - Creating FTO's from our internship tasklist

### ISSUE THREAD

**Issue by Tlazypanda** [publiclab/plots2#8280]
Hi everybody! 👏

We all at Public Lab 🎈 - learn, grow, work, brainstorm ideas, contribute together so why not share about our weekly goals and the awesome work we have done at Public Lab with each other, so we can support and collaborate with each other better. We have a Community Check-In each week, where every community member can share something about their work from the past week and about their current week's goal 🎯 . You are also welcome to share fun-fact 😄 , new ideas 💡 , your learning goals ☑️.

We believe in collaborative efforts to support our community. We are running a learning platform which helps a newcomer to become a leader of tomorrow. 💯

## Creating FTO's from our internship tasklist

Hey everyone! Since we are nearing the end of our internships, let's take this week to focus on finding some tweaks/minor improvements that we had planned but were unfortunately unable to take up and then breaking down those issues into ftos. We can also create fto's out of future improvements that we would like to be taken up for our project or something that promotes further maintainability of our code :tada: 

A good step to start would be to go through the current tasklist and update the entries and then defining a way to break down all the remaining tasks to allow new contributors to easily contribute to them. If you are finding difficulty in breaking down since the tasks are relatively more difficult ,you may also create some more experienced issues and label as help wanted or fto-candidates for our more experienced contributors to take on!  Let's carry on the legacy of our work :joy: 

### For interns
Please also share issues you're working on that you need some help with. This might also be a good time to take a step back and define the scope of our projects 

## Next Check-in
If you would like to open the next check-in..Leave a comment below. We are happy to help if its your first time

**Thanks everyone for making Public Lab awesome** 🎈
Have a great week and year ahead all ❤️

**Comment by Tlazypanda** [comment]
@jywarren, @AgyeyaMishra, @Jolios, @cesswairimu, @samarakoon-ryan, @Samasth, @malaikaaiyar, @keshavsethi, @OmLakshmi, @nsachin08, @dependabot-preview[bot], @Tlazypanda, @galanggg, @ishika1727, @icarito, @RajatSablok, @mronauli, @ijeomaemeruwa, @alaxalves, @Shulammite-Aso, @JapneetSingh5, @Neilblaze, @markkulube, @vinuch @JapneetSingh5, @Neilblaze, @markkulube, @vinuch, @Tlazypanda, @icarito, @dependabot-preview[bot], @jywarren, @keshavsethi, @praaatik, @Janitham97, @imanishbarnwal, @dburugupalli, @adityakpranav, @ioampatzidis, @esiebomaj, @Alexjake660, @cesswairimu, @urvashigupta7, @daemon1024, @jscastanos @keshavsethi, @jywarren, @AyanChoudhary, @imajit,@NitinBhasneria, @nstjean, @marientje, @rutger-t, @Tlazypanda,

**Comment by Tlazypanda** [comment]
@llabake, @khushi02, @Alexjake660, @LarrySul, @nicoleiocana, @chrisdrums1, @Angelvicks, @Akanksha1212, @beladinaelfitri, @rid1hady, @shreyaa-sharmaa, @goamoth, @SusannaDiV, @yusofsf, @nanlliu, @Tlazypanda,@keshav234156, @WaldenJosh, @B4Lee, @VisnuGanth, @jessibreen, @alaxalves, @kamal-kaur04, @sssash18, @chinmay021, @Srija616, @AnnaMinakova, @greenbrown, @taysa91, @TaishiRoy, @yagmurbaydogan, @Shulammite-Aso, @Jigyasa-Kumari @moliver4, @justinelai, @Snouzy, @stevenalila, @Digiox, @yandu31, @milosv1, @jamescapps, @Nevenem, @chirag-wadhwa, @yogeshjdeshmukh, @omizrahi99, @theOpetunde, @sailahonkanen, @robin-natale, @avidler, @ccyccyccy

**Comment by Tlazypanda** [comment]
@Janet33, @TroydonAnabolic, @priyanshbalyan, @Medlineme4, @mandysGit, @msanden, @AidenOliver, @Janetsomehow, @AntroMarina, @llabake, @khushi02, @Alexjake660, @LarrySul, @nicoleiocana, @chrisdrums1, @Angelvicks, @Akanksha1212, @beladinaelfitri, @rid1hady, @shreyaa-s, @goamoth, @SusannaDiV, @yusofsf, @nanlliu, @WaldenJosh, @B4Lee, @VisnuGanth, @jessibreen, @alaxalves, @kamal-kaur04 @adityakpranav @esiebomaj @Yonttee @swarajpure @vinuch @ishika1727 @nsachin08 @Rohit-786 @daemon1024 @jscastanos, @sarthak-choudhary, @dongskyler, @keshavsethi, @Shulammite-Aso, @ckchu8, @keshav234156, @DerekChristy, @Srija616, @priyanka-choubey, @AyanChoudhary, @imajit,

**Comment by Tlazypanda** [comment]
@jscastanos, @sarthak-choudhary, @dongskyler, @keshavsethi, @Shulammite-Aso, @ckchu8, @keshav234156, @DerekChristy, @Srija616, @priyanka-choubey, @AyanChoudhary, @imajit, @NitinBhasneria, @nstjean, @marientje, @rutger-t, @imajit, @NitinBhasneria, @nstjean, @marientje, @rutger-t, @nsachin08 @Rohit-786 @Tristan-Elizabeth @aakanchha01 @habeckmatt @hiteshsubnani0128 @ijeomaemeruwa @OmLakshmi @dmoini @yusufatalay @Samasth @malaikaaiyar @davidgoldcode @Shokescodes @marciojunior88 @avats-dev @NitinBhasneria, @nstjean, @marientje, @rutger-t, @urvashigupta7, @daemon1024

**Comment by sagarpreet-chadha** [comment]
Great check-in and excellent idea 🎉

**Comment by alaxalves** [comment]
Sorry for being missing a little this past week. Unfortunately a close friend of mine died in a motorcycle accident and my family and I have been taking care and supporting the loved ones. Hopefully God will bless them with some comfort in these hard times.
I'll focus now on implementing the system tests on Spectral Workbench and implement GitPod on the Exporter Sinatra repo.

**Comment by jywarren** [comment]
Sending ❤️ Alax.

**Comment by jywarren** [comment]
Editor team - some great updates.

@Shulammite-Aso @keshav234156 @NitinBhasneria @shreyaa-sharmaa great work!

@NitinBhasneria and @sagarpreet-chadha got the `v2.1.1` fix completed - it's now published with the patch we hope will get it passing on plots2 https://github.com/publiclab/PublicLab.Editor/issues/587 and https://github.com/publiclab/PublicLab.Editor/issues/588. Now we're converting the next release into `v2.1.2` and compiling release notes from all your work since the last release. Please chime in with any updates to be included in the release notes! Thanks! 

https://github.com/publiclab/PublicLab.Editor/pull/557/

Really excited to be unblocked there! Fingers X and congrats, all. 🎉 🎉

**Comment by jywarren** [comment]
Whew, ok it's running against plots2 tests here: https://github.com/publiclab/plots2/pull/8285

**Comment by jywarren** [comment]
@Tlazypanda i love this theme. You've been a real leader in this regard, truly welcoming lots of people in and growing our community. Thank you, and i think it's a GREAT idea for folks to start!

**Comment by Tlazypanda** [comment]
@alaxalves sorry for your loss may these testing times pass soon ❤️

**Comment by Tlazypanda** [comment]
Congratulations to the editor team 🎉 

Thanks @jywarren  just working towards the core values of public lab learning from all the other mentors and interns here 😅 

This week I worked on some more performance optimizations (please review whenever free 😅 #8272 #8269 #8268 #8267 #8183 ) and also updated my tasklist finally after procrastinating for so long 😂 #7883 Realised that still have many improvements to be done but hopefully will continue after the end slowly and eventually get there

**Comment by jywarren** [comment]
Hi all, new editor `v2.1.2` is now live at https://stable.publiclab.org/post ! Please try it out and confirm that your recent changes are working :-)))))

**Comment by jywarren** [comment]
Aha - found an issue already - no worries, we can patch it! @NitinBhasneria i think it's related to the main image upload feature! https://github.com/publiclab/PublicLab.Editor/issues/591

**Comment by cesswairimu** [comment]
Hi everyone...update...This week's focus issue is https://github.com/publiclab/PublicLab.Editor/issues/591. We  need this for the  Editor release o be merged in plots2 and will appreciate you help. Thanks in advance

**Comment by sagarpreet-chadha** [comment]
Thanks @cesswairimu for highlighting that issue 💯 

@publiclab/editor 
I am mentioning the steps that we have to follow before publishing next releases:
1. Increment the version in package.json from x.y.z to:

> x+1.y.z if there is a major changes like new features or change in syntax, etc.

> x.y+1.z if there is a minor change like changing in some logic or upgrading dependent libraries maybe, etc

> x.y.z+1 if there are some patches to be made like some bug is solved, etc

2. Make sure `package-lock.json` is changed if some dependencies are changed in package.json.

3. Make sure your changes are there in `dist` folder.

4. That's it for PL.editor repo! Let's publish to npm via `npm publish` command.
PS: this will publish the `dist` folder from your local and not from github.

5. Once published to `npm`, dependabot will make PR automatically in `plots2` to upgrading the PL editor version in package.json.

6. Assuming that proper versioning is done in PL.editor, then for minor and patch releases all checks will pass 💯 
But for major release, the checks might fail which is normal 👍 .

7. For major releases, check into the dependabot branch or make a new PR having package upgrade changes.
After manually changing version in package.json, make sure to run `npm install` -> this will install the latest PL editor to your local `node_modules` folder.
Now make sure that the breaking change is also added to your PR like syntax change, or if `id` of some `div` is changed in PL.editor then same change has to be done in
`rich.html.erb` file.

8. All done now 🎉 , make sure the PR ia reviewed and then the integration is completed 😄 .

**Comment by jywarren** [comment]
Great, we merged the fix to stable server and i'm tracking here: publiclab/plots2#8286 -- good work Nitin. Technically this has been a breaking change though; in the future let's do our utmost to avoid a requirement that downstream projects change their usage to accommodate new requirements from our library, unless we're ready to propose a full version number. 

In this case we are the only user of the lib, so we don't have to be too stressed, but technically this would have been a v3.0.0 release, does that make sense?

Awesome work and congrats, folks!

**Comment by ebarry** [comment]
Hi all, I'm looking ahead at the Public Lab Open Hour in September where each and every Outreachy and Google Summer of Code participant -- @Shulammite-Aso @shreyaa-sharmaa @Tlazypanda @NitinBhasneria @keshav234156 @keshavsethi @alaxalves -- will share 5 minutes about their summer's work! 

- Based on input from last week's check-in, we are planning to use Zoom, and set up a meeting for at 8AM Pacific US West Coast /  8:30pm India Standard Time -- see https://everytimezone.com/?t=5f557800,384 
- Also based on input from last week's check-in, we will not meet on Tuesday September 8, but instead look at Wednesday September 9. 

**Does this date and time work for you?** 

To practice, and also in case of any bandwidth problems, consider recording a screencapture of your presentation ahead of time. 

Please let me know how this would work for you, i need to lock this in in the next day. Thank you!

**Comment by cesswairimu** [comment]
Moved to https://github.com/publiclab/community-toolbox/issues/386. Thanks everyone
